src/internal/partial_solution.rs

```diff
@@ -30,24 +30,43 @@ impl DecisionLevel {
 #[derive(Clone, Debug)]
 pub(crate) struct PartialSolution<DP: DependencyProvider> {
     next_global_index: u32,
+    /// The number of decisions that have been made, equal to the number of packages with decisions.
     current_decision_level: DecisionLevel,
-    /// `package_assignments` is primarily a HashMap from a package to its
-    /// `PackageAssignments`. But it can also keep the items in an order.
-    ///  We maintain three sections in this order:
-    /// 1. `[..current_decision_level]` Are packages that have had a decision made sorted by the `decision_level`.
-    ///    This makes it very efficient to extract the solution, And to backtrack to a particular decision level.
-    /// 2. `[current_decision_level..changed_this_decision_level]` Are packages that have **not** had there assignments
-    ///    changed since the last time `prioritize` has been called. Within this range there is no sorting.
-    /// 3. `[changed_this_decision_level..]` Contains all packages that **have** had there assignments changed since
-    ///    the last time `prioritize` has been called. The inverse is not necessarily true, some packages in the range
-    ///    did not have a change. Within this range there is no sorting.
+    /// Store for all known package decisions and package derivations.
+    ///
+    /// "assignment" refers to both packages with decisions and package with only derivations and
+    /// no decision yet.
+    ///
+    /// `[..current_decision_level]`: Packages that have had a decision made, sorted by the
```

Comment:
Can you provide a little more context before jumping into these ranges? Like set the stage for the format of this commentary?